### DISCUSSION TITLE
Element always opens in fullscreen

### DISCUSSION THREAD

**Discussion by jpmvferreira** [i3/i3#4746] [Q&A] [answered]
Hi there, this is probably not an issue related with i3, but with Element itself.

As the title says, every time I open Element it opens in fullscreen. This also used to happen for Atom for a while but, after an update, it eventually got fixed.

The reason why I'm opening up a discussion here is because I'm out of ideas on how to diagnose this problem and make a proper bug report to Element, or just find a quick fix that does the job. All I've tried was setting fullscreen disable in the i3 config for this specific class of windows, but it didn't work.

Any ideas on how to approach this problem?

**Comment by ghost** [comment, marked as answer]
I don't know what "Element" is but you can try something like this. You can use any option for ``for_window``.

* 1st. Find proper ``WM_CLASS(STRING)`` name.
* 2nd. Edit your i3wm config file with something like;
`` for_window [class="CLASS-NAME-HERE"] floating enable, border toggle, resize set 500px 500px, move position 50px 50px``

**Reply by jpmvferreira** [reply]
I've tried something similar, which was `for_window [class="Element"] fullscreen disable` that doesn't really do anything. 
When using your suggestion it still opens in fullscreen. Only when I disable the fullscreen it becomes floating.
So it seems to be refusing to open without fullscreen.

**Reply by Airblader** [reply]
I suspect that it doesn't open fullscreen when the window is mapped, but gets fullscreened after the fact. Your config file and a log file would be necessary to investigate that.

**Reply by jpmvferreira** [reply]
Here's my i3 config file:
[config.txt](https://github.com/i3/i3/files/7776364/config.txt)

Regarding the log I'm not quite sure what I'm looking for and how to get the relevant section of the logs, so what I did was running `i3-dump-log`, open element, disable the fullscreen and run `i3-dump-log` again.
So here's that part of the logs:
[batata.txt](https://github.com/i3/i3/files/7776367/batata.txt)

I forgot to mention, since it was asked, Element is an electron based (just like Atom, that's why I consider that detail relevant) Matrix client. In fact it's the "standard" Matrix client, being also developed by the same organization which develops the Matrix protocol.

**Reply by Airblader** [reply]
> I'm not quite sure [...] how to get the relevant section of the logs

Start a new i3 session, have nothing open, just immediately start Element, pull the logs, do not remove any lines.

**Reply by jpmvferreira** [reply]
Okay so here's what I did, I had i3 open, I switched to another TTY, edited my `.xinitrc` to include `exec i3` because lightdm is in charge of starting my WM, ran `startx` on the TTY, opened up element (using rofi) changed to another workspace, opened up a terminal (Kitty) and typed `i3-dump-log` twice, here's the output:
```console
$ i3-dump-log
i3-dump-log: i3 is running, but SHM logging is not enabled. Enabling SHM log now while i3-dump-log is running

26/12/2021 22:10:27 - ../i3-4.20.1/src/commands.c:cmd_criteria_init:148 - Initializing criteria, current_match = 0x5621e16e1d60
26/12/2021 22:10:27 - ../i3-4.20.1/src/handlers.c:handle_event:1380 - event type 28, xkb_base 85

$ i3-dump-log
i3-dump-log: i3 is running, but SHM logging is not enabled. Enabling SHM log now while i3-dump-log is running

26/12/2021 22:10:48 - ../i3-4.20.1/src/commands.c:cmd_criteria_init:148 - Initializing criteria, current_match = 0x5621e16e1d60
26/12/2021 22:10:48 - ../i3-4.20.1/src/handlers.c:handle_event:1380 - event type 28, xkb_base 85
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - dock status does not match
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - dock status does not match
26/12/2021 22:10:48 - ../i3-4.20.1/src/assignments.c:run_assignments:18 - Checking if any assignments match this window
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression ".*" matches "kitty"
26/12/2021 22:10:48 - window class matches (kitty)
26/12/2021 22:10:48 - dock status matches
26/12/2021 22:10:48 - ../i3-4.20.1/src/assignments.c:run_assignments:33 - This assignment already ran for the given window, not executing it again.
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "File Transfer*" does not match "undercover@Legion:~"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Tk" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Nm-connection-editor" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Galculator" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Lightdm-settings" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Lxappearance" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Manjaro Settings Manager" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Nitrogen" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "matplotlib" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Oblogout" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Simple-scan" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "(?i)System-config-printer.py" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Timeset-gui" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Xfce4-power-manager-settings" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Thunderbird" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Thunderbird" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Thunderbird" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Tlp-UI" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Gnome-calculator" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "qt5ct" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "jetbrains-clion" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Matplotlib" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Kvantum Manager" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "uk-ac-starlink-topcat-Driver" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "flameshot" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Spotify" does not match "kitty"
26/12/2021 22:10:48 - Checking window 0x0280000f (class kitty)
26/12/2021 22:10:48 - Regular expression "Element" does not match "kitty"
26/12/2021 22:10:48 - ../i3-4.20.1/src/x.c:x_push_changes:1198 - -- PUSHING WINDOW STACK --
26/12/2021 22:10:48 - ../i3-4.20.1/src/x.c:x_push_changes:1273 - PUSHING CHANGES
26/12/2021 22:10:48 - ../i3-4.20.1/src/x.c:x_push_changes:1366 - ENDING CHANGES
26/12/2021 22:10:48 - ../i3-4.20.1/src/handlers.c:handle_event:1380 - event type 28, xkb_base 85
26/12/2021 22:10:48 - ../i3-4.20.1/src/handlers.c:handle_event:1380 - event type 28, xkb_base 85
```

Does this make sense? it only includes two lines in the first log dump although it should initialize the WM from the ground up, or is it better if I reboot my machine, login using lightdm, open up Element and then dump the log?

**Reply by Airblader** [reply]
> Does this make sense?

No, not really. The i3 log will start with i3's startup process, which includes for example dumping the loaded configuration. You can refer to the [debugging docs](https://i3wm.org/docs/debugging.html) for some information.

**Reply by jpmvferreira** [reply]
Here it is, I've rebooted my machine, logged in with i3 in debug mode, opened up element, changed workspace, opened up kitty and dumped the log to a file:
[log.txt](https://github.com/i3/i3/files/7833944/log.txt)

**Reply by Airblader** [reply]
```
08/01/2022 20:05:00 - ../i3-4.20.1/src/handlers.c:handle_event:1380 - event type 33, xkb_base 85
08/01/2022 20:05:00 - ClientMessage for window 0x02200001
08/01/2022 20:05:00 - ../i3-4.20.1/src/handlers.c:handle_client_message:685 - toggling fullscreen
```

The application sends a client message to tell the window manager that it would like to be fullscreen (`_NET_WM_STATE_FULLSCREEN`). This happens _after_ your assignment has been executed. If the application has no configuration that prevents this, you'd have to either do something like `sleep 1 && [class="Element"] fullscreen disable` (which is pretty ugly and unreliable), or write a script that watches for this situation to execute it (still ugly). But the "fault" here lies with the application, i3 just executes what the application requests.

---

Side note:

```
for_window [class=".*"] border pixel 3   # gotta place this here because without when the border resets to none for some reason
```

produces an error, you should move the comment to its own line.

**Reply by jpmvferreira** [reply]
Thanks, I'll mark the top level discussion as the answer, because I'm not allowed to mark the previous comment as the solution.

I guess I'll report this to Element desktop then, now with further confidence that it isn't an issue that I caused on something related to i3WM.

Regarding the side note, I'm not quite sure why I have written that line. When you say it produces an error does it mean that it doesn't get executed or that it gets executed but it produces an error because comments should be in a separate line?
